test/active_support/cache/redis_store_test.rb

```diff
@@ -508,12 +514,25 @@ def with_store_management
 
     def with_notifications
       @events = [ ]
-      ActiveSupport::Cache::RedisStore.instrument = true
-      ActiveSupport::Notifications.subscribe(/^cache_(.*)\.active_support$/) do |*args|
-        @events << ActiveSupport::Notifications::Event.new(*args)
+      instrument_store do
+        ActiveSupport::Notifications.subscribe(/^cache_(.*)\.active_support$/) do |*args|
+          @events << ActiveSupport::Notifications::Event.new(*args)
+        end
+        yield
+      end
+    end
+
+    if ActiveSupport::VERSION::MAJOR < 5
```

Comment:
why not make `def instrument_store` do the conditional inside the method so you're not conditionally defining a method?